cmd/agent/agent.go

```diff
@@ -250,20 +250,33 @@ func startManager(o *options.AgentOptions, ctx context.Context) {
 			Agent:                    agent,
 		}
 		clusterClaimer := clusterclaimctl.ClusterClaimer{
-			ClusterName:                     o.ClusterName,
-			HubClient:                       mgr.GetClient(),
-			KubeClient:                      managedClusterKubeClient,
-			ConfigV1Client:                  openshiftClient,
-			OauthV1Client:                   osOauthClient,
-			Mapper:                          reloadMapper,
-			EnableSyncLabelsToClusterClaims: o.EnableSyncLabelsToClusterClaims,
+			KubeClient:     managedClusterKubeClient,
+			ConfigV1Client: openshiftClient,
+			OauthV1Client:  osOauthClient,
+			Mapper:         reloadMapper,
 		}
 
-		clusterClaimReconciler := clusterclaimctl.ClusterClaimReconciler{
-			Log:               ctrl.Log.WithName("controllers").WithName("ManagedClusterInfo"),
-			ClusterClient:     managedClusterClusterClient,
-			ClusterInformers:  clusterInformerFactory.Cluster().V1alpha1().ClusterClaims(),
-			ListClusterClaims: clusterClaimer.List,
+		clusterClaimReconciler, err := clusterclaimctl.NewClusterClaimReconciler(
+			ctrl.Log.WithName("controllers").WithName("ClusterClaim"),
+			managedClusterClusterClient,
+			clusterInformerFactory.Cluster().V1alpha1().ClusterClaims().Lister(),
+			clusterClaimer.GenerateExpectClusterClaims,
+		)
+		if err != nil {
+			setupLog.Error(err, "unable to create controller", "controller", "ClusterClaim")
+			os.Exit(1)
+		}
+
+		labelClusterClaimReconciler, err := clusterclaimctl.NewLableClusterClaimReconicler(
```

Comment:
I don't think we need create another reconciler controller, we can use different sync functions in one controller like the common implementation  in ocm repo.

Reply:
Thanks for the advise, do we means multiple `syn`c in one controller here? But I didn't find the practice in the `ocm` repo, would you mind help me with an example?

Also worth to mention, a motivation of separating a new controller is using different eventhandlers to filter clusterclaim events. For example, if a `spoke-only` clusterclaim updates, we only want to trigger the reconcile of `labelClusterClaimReconciler`.

Reply:
Thanks @zhiweiyin318 for providing the example: https://github.com/open-cluster-management-io/ocm/blob/5405099cb5b8e49a547f8e3771f65087fd7c0bfd/pkg/addon/controllers/addonmanagement/controller.go#L105

This is a good practice of simply the code, @qiujian16 would you mind taking a second look? I think as zhiwei said, this could also be a way of 'sperate controller'?

Reply:
Confirmed with @qiujian16, we have no preference on this, both ways are ok, I will change to use one controller to follow the same style with other codes.